Issue 41 cimulator buid

alex124585 opened the pull request:
Add test which check that c simulator is built correctly

ZichangHe reviewed (approved):
Looks good!

rsln-s commented on setup.py:
```diff
@@ -2,47 +2,27 @@
 # // SPDX-License-Identifier: Apache-2.0
 # // Copyright : JP Morgan Chase & Co
 ###############################################################################
+# from setuptools import setup, Extension
```
Comments should be removed (duplicating the line below?)

rsln-s commented on setup.py:
```diff
@@ -2,47 +2,27 @@
 # // Copyright : JP Morgan Chase & Co
 ###############################################################################
+# from setuptools import setup, Extension
 from setuptools import setup, Extension
```
Imports are now unused?

rsln-s commented on setup.py:
```diff
@@ -2,47 +2,27 @@
-            super().run
-        except Exception as e:
-            print("No C/C++ enviroment setup to compile the C simulator. Installing Python Simulator")
+if python_only is not None:
```
Can this condition be used directly on line 23 below instead of creating an intermediary `QOKIT_PYTHON_ONLY` variable?

alex124585 replied:
Need that env variable to control workflow/debug

rsln-s replied:
Looks like more tests are failing now...

alex124585 replied:
> Looks like more tests are failing now...

It should right? This is python build only no c-build

rsln-s replied:
Why did the tests start failing though? Responding in this thread, but not sure which change triggered it

rsln-s replied:
If it's intended, it should catch the error, no?

rsln-s commented on tests/test_simulators_builds.py:
```diff
@@ -0,0 +1,35 @@
+from qokit import get_qaoa_labs_objective
+from qokit.fur import get_available_simulator_names
+
+# when fales tests only runs in Github actions. Chage to true to run locally
```
Can you please fix the typos? 

"fales" -> "fails"?

"Chage" -> "change"?

rsln-s reviewed (changes requested):
A few minor comments. Please also note that the workflow should succeed in Github actions (currently failing).

rsln-s commented:
Looks like more tests are failing now...

rsln-s commented on tests/test_simulators_builds.py:
```diff
@@ -0,0 +1,35 @@
+from qokit.fur import get_available_simulator_names
+
+# when fail tests only runs in Github actions. Change to true to run locally
+IN_GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS") == "fales"
```
Typo! "fales" -> "fails"

rsln-s commented on tests/test_simulators_builds.py:
```diff
@@ -0,0 +1,35 @@
+IN_GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS") == "fales"
+
+
+@pytest.mark.skipif(IN_GITHUB_ACTIONS, reason="Test runs only in Github Actions.")
```
TODO: add graceful handling of this in python pipeline

rsln-s commented on .github/workflows/pypi-test-publish.yml:
```diff
@@ -11,7 +11,8 @@ jobs:
     strategy:
       matrix:
-        os: [ubuntu-22.04, macos-11, windows-2022]
+        #os: [ubuntu-22.04, macos-11, windows-2022]
```
TODO: add back macOS and (possibly) windows

rsln-s reviewed (changes requested):
The tests are still failing. Please make sure that the tests pass before requesting re-review!

alex124585 commented:
> The tests are still failing. Please make sure that the tests pass before requesting re-review!

It was not failing. Issue is we have 2 pipeline for python only and c/c++ running in parallel that is what create this intermittent failure.

> The tests are still failing. Please make sure that the tests pass before requesting re-review!

tests are not failing as you can see in github actions issue is that 2 workflows for c/c++ and pure python works in parallel which creates this intermittent error need to run jobs sequentially

rsln-s commented:
Status update: there are two main problems
(i) the PYTHON_ONLY flag is not being raised properly in "python only" workflow: 
https://github.com/jpmorganchase/QOKit/actions/runs/10218741603/job/28275531066?pr=47
(ii) the simulator is not being built properly on MacOS:
https://github.com/jpmorganchase/QOKit/actions/runs/10218741601/job/28275532498?pr=47

rsln-s commented:
Update: the Mac build failure is due to clang not supporting openmp. Installing GCC should resolve the issue, but I can't get make to use the right compiler

alex124585 commented:
fixed following pipeline issues:
1. Mac OS was added with gcc brew install but in Mac OS default path is to Apple clang
2.  Added correct path for gcc - export PATH=$PATH:/opt/homebrew/
3. Corrected test case for python only
4. Removed incorrectly added  make in pipeline. No need for make since make is in setup.py
